lib/service/service.go

```diff
@@ -898,6 +904,26 @@ func NewTeleport(cfg *Config) (*TeleportProcess, error) {
 		cloudLabels.Start(supervisor.ExitContext())
 	}
 
+	var cmdLabels *labels.Dynamic
+	var cmdLabelKeys map[string]uint64
+	if len(cfg.CmdLabels) != 0 {
+		cmdLabels, err = labels.NewDynamic(supervisor.ExitContext(), &labels.DynamicConfig{
```

Comment:
The command label values are cloned in `labels.NewDynamic` to prevent mutating the contents of the config file.

Reply:
Ah... perf-wise I don't think it matters.  Hadn't considered that users might have put something with side-effects in there though.

That's a tricky problem.  On one hand, given the fact that we use stuff like `Sync` and make no effort to preserve the periodicity across restarts, the command run period is already sort of an upper bound... and using commands for side-effect stuff is misuse IMO... but *if* someone is using it like a poor man's cron job, doubling the executions would be bad.

I'm gonna have to think about this one... I dislike the idea of adding too much interconnected machinery between instance and ssh server state, but might be necessary 😬

Reply:
Alright, tried a few different strategies.  Ended up just having `service/service.go` use a shared `labels.Dynamic` if instance-level and service-level `CmdLabels` are equal.

Technically, this will still interfere with users using commands for their side-effects if they just happen to define identical commands at both the service and instance level, but IMO that's less problematic, and being smarter about it (only sharing the labels if labels were actually only defined in one place) started to get a bit convoluted.